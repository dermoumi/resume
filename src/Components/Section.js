import React, { Component } from 'react';
import styled from 'styled-components';

const StyleBase = styled.section`
  margin-top: 32px;
  margin-bottom: 54px;

  &:last-child {
    margin-bottom: 16px;
  }

  h3 {
    margin-top: 0;
    margin-bottom: 12px;
    font-size: 1em;
    font-weight: 600;
    text-transform: uppercase;
    position: relative;
    display: inline-block;
    color: ${props => props.theme.primary};
  }

  a {
    color: ${props => props.theme.primary};
  }
`;

export default class Section extends Component {
  render() {
    return (<StyleBase {...this.props}>
      <h3>{this.props.label}</h3>
      <div className="section-content">
        {this.props.children}
      </div>
    </StyleBase>);
  }
}
