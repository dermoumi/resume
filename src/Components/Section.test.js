import React from 'react'
import { mount, shallow } from 'enzyme'
import 'jest-styled-components'

import Section from './Section'

it('renders the section label', () => {
  const label = 'My label'
  const wrapper = shallow(<Section label={label} />)

  expect(wrapper.text()).toContain(label)
})

it('renders the section content', () => {
  const contentText = 'My content text'
  const content = <div>{contentText}</div>
  const wrapper = shallow(<Section>{content}</Section>)

  expect(wrapper.text()).toContain(contentText)
})

it('respects the theme', () => {
  const theme = {
    primary: '#FFCC00',
  }

  const props = {
    label: 'My title',
    children: <div>Content</div>,
    theme,
  }

  const wrapper = mount(<Section {...props} />)

  expect(wrapper).toHaveStyleRule('color', theme.primary, {
    modifier: 'h3',
  })

  expect(wrapper).toHaveStyleRule('color', theme.primary, {
    modifier: 'a',
  })
})
