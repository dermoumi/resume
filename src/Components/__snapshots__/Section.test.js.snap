// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component respects the theme 1`] = `
.c0 {
  margin-bottom: 38px;
}

.c0:last-child {
  margin-bottom: 16px;
}

.c0 h3 {
  margin-top: 0;
  margin-bottom: 12px;
  font-size: 1em;
  font-weight: 600;
  text-transform: uppercase;
  position: relative;
  display: inline-block;
  color: #FFCC00;
}

.c0 a {
  color: #FFCC00;
}

@media print {
  .c0 {
    margin-bottom: 76px;
  }
}

<Section
  label="My title"
  theme={
    Object {
      "primary": "#FFCC00",
    }
  }
>
  <styled.section
    label="My title"
    theme={
      Object {
        "primary": "#FFCC00",
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  margin-bottom: 38px;

  &:last-child {
    margin-bottom: 16px;
  }

  h3 {
    margin-top: 0;
    margin-bottom: 12px;
    font-size: 1em;
    font-weight: 600;
    text-transform: uppercase;
    position: relative;
    display: inline-block;
    color: ",
              [Function],
              ";
  }

  a {
    color: ",
              [Function],
              ";
  }

  @media print {
    margin-bottom: 76px;
  }
",
            ],
          },
          "displayName": "styled.section",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      label="My title"
      theme={
        Object {
          "primary": "#FFCC00",
        }
      }
    >
      <section
        className="c0"
        label="My title"
      >
        <h3>
          My title
        </h3>
        <div
          className="section-content"
        >
          <div>
            Content
          </div>
        </div>
      </section>
    </StyledComponent>
  </styled.section>
</Section>
`;
